meteor-app/imports/ui/assessment/assessment-add.js

```diff
@@ -217,41 +198,48 @@ class AssessmentAdd extends Component {
       .map(key => {
         return key.name
       })
-    this.setState({
+    this.setState((prevState) => ({
       formData: {
         ...formData,
         services: majorServices,
         package: "Major Service Package"
       },
-      step: this.state.step + 1,
-      progress: this.state.progress + 1
-    })
+      step: prevState.step === prevState.progress || prevState.step < prevState.progress  ? this.state.step + 1 : prevSate.step,
```

Comment:
This code looks like it appears in many place - a good candidate for being refactored into a common method